src/main/java/com/spotify/dns/AggregatingChangeNotifier.java

```diff
@@ -30,6 +30,7 @@
   private final List<ChangeNotifier<T>> changeNotifiers;
 
   private volatile Set<T> records = ImmutableSet.of();
+  private volatile boolean waitingForFirstEvent = true;
```

Comment:
No specific `Atomic` operations are needed. I guess it could use some `compareAndSet` instead actually, is that what you meant?